error: unused variable ‘_’ [-Werror=unused-variable]

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.


### Version

kvrocks：2.7.0
OS：CentOS Linux release 7.9.2009 (Core)
gcc：gcc version 11.2.1 20220127 (Red Hat 11.2.1-9) (GCC)
cmake：cmake version 3.26.0-rc6


### Minimal reproduce step

./x.py build -DDEPS_FETCH_PROXY=https://hub.nuaa.cf/

### What did you expect to see?

希望能解决兼容问题，实现正常编译

### What did you see instead?

![1](https://github.com/apache/kvrocks/assets/23118518/8df5db4c-e837-48bb-9220-6650cf202213)
[ 84%] Linking C static library lib/libevent.a
[ 84%] Built target event_static
[ 84%] Building CXX object _deps/fmt-build/CMakeFiles/fmt.dir/src/format.cc.o
[ 84%] Building CXX object _deps/fmt-build/CMakeFiles/fmt.dir/src/os.cc.o
[ 84%] Linking CXX static library libfmt.a
[ 84%] Built target fmt
[ 85%] Building CXX object CMakeFiles/kvrocks_objs.dir/src/cluster/cluster.cc.o
/root/apache-kvrocks-2.7.0-src/src/cluster/cluster.cc: In member function ‘std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> > Cluster::getClusterNodeSlots() const’:
/root/apache-kvrocks-2.7.0-src/src/cluster/cluster.cc:537:22: error: unused variable ‘_’ [-Werror=unused-variable]
   for (auto &[_, info] : slots_infos) {
                      ^
cc1plus: some warnings being treated as errors
gmake[3]: *** [CMakeFiles/kvrocks_objs.dir/build.make:76: CMakeFiles/kvrocks_objs.dir/src/cluster/cluster.cc.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:530: CMakeFiles/kvrocks_objs.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:577: CMakeFiles/kvrocks.dir/rule] Error 2
gmake: *** [Makefile:254: kvrocks] Error 2
Traceback (most recent call last):
  File "./x.py", line 422, in <module>
    args.func(**arg_dict)
  File "./x.py", line 137, in build
    run(cmake, *options, verbose=True, cwd=dir)
  File "./x.py", line 68, in run
    raise RuntimeError(err)
RuntimeError: 
failed to run: ('/usr/local/bin/cmake', '--build', '.', '-t', 'kvrocks', 'kvrocks2redis')
exit with code: 2

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -212,7 +212,16 @@ add_library(kvrocks_objs OBJECT ${KVROCKS_SRCS})
 target_include_directories(kvrocks_objs PUBLIC src src/common src/vendor ${PROJECT_BINARY_DIR} ${Backtrace_INCLUDE_DIR})
 target_compile_features(kvrocks_objs PUBLIC cxx_std_17)
 target_compile_options(kvrocks_objs PUBLIC -Wall -Wpedantic -Wsign-compare -Wreturn-type -fno-omit-frame-pointer)
-target_compile_options(kvrocks_objs PUBLIC -Werror=unused-result -Werror=unused-variable)
+target_compile_options(kvrocks_objs PUBLIC -Werror=unused-result)
+
+# disable unused-variable check on GCC < 8 due to the structure bindings
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?format=multiple&id=81767
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8)
+    target_compile_options(kvrocks_objs PUBLIC -Wno-error=unused-variable)
+else()
+    target_compile_options(kvrocks_objs PUBLIC -Werror=unused-variable)
+endif()
+
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     target_compile_options(kvrocks_objs PUBLIC -Wno-pedantic)
 elseif((CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"))

--- a/src/server/server.cc
+++ b/src/server/server.cc
@@ -1633,7 +1633,7 @@ StatusOr<std::unique_ptr<redis::Commander>> Server::LookupAndCreateCommand(const
   auto cmd = cmd_attr->factory();
   cmd->SetAttributes(cmd_attr);
 
-  return cmd;
+  return std::move(cmd);
 }
 
 Status Server::ScriptExists(const std::string &sha) {

